srtcore/sync.cpp

```diff
@@ -659,4 +665,70 @@ bool srt::sync::StartThread(CThread& th, void* (*f) (void*), void* args, const c
     return true;
 }
 
+////////////////////////////////////////////////////////////////////////////////
+//
+// CThreadError class - thread local storage error wrapper
+//
+////////////////////////////////////////////////////////////////////////////////
+namespace srt {
+namespace sync {
+
+class CThreadError
+{
+public:
+    CThreadError()
+    {
+        pthread_key_create(&m_TLSError, TLSDestroy);
+    }
+
+    ~CThreadError()
+    {
+        delete (CUDTException*)pthread_getspecific(m_TLSError);
```

Comment:
This is kinda useless because this will be deleted anyway at the end of the thread that is exiting through `TLSDestroy` registered with `pthread_key_create()`. If you really want to do this, this is also wrong because when you delete the object, you should also set the current value to NULL by `pthread_setspecific`. Otherwise you'll get double delete.